core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -235,16 +236,18 @@ public String toString() {
   private final List<SortOrder> sortOrders;
   private final Map<String, String> properties;
   private final long currentSnapshotId;
-  private final List<Snapshot> snapshots;
-  private final Map<Long, Snapshot> snapshotsById;
   private final Map<Integer, Schema> schemasById;
   private final Map<Integer, PartitionSpec> specsById;
   private final Map<Integer, SortOrder> sortOrdersById;
   private final List<HistoryEntry> snapshotLog;
   private final List<MetadataLogEntry> previousFiles;
-  private final Map<String, SnapshotRef> refs;
   private final List<StatisticsFile> statisticsFiles;
   private final List<MetadataUpdate> changes;
+  private final SerializableSupplier<List<Snapshot>> snapshotsSupplier;
+  private List<Snapshot> snapshots;
+  private Map<Long, Snapshot> snapshotsById;
+  private Map<String, SnapshotRef> refs;
+  private boolean snapshotsLoaded;
```

Comment:
Can you default this in the constructor?

Also, I think it would make sense to set it to `true` if `snapshotsSupplier` is null. That way it is clear that if there is no supplier the snapshot list should be complete.